add configfs implementation

jostillmanns opened the pull request:
`pligos helmfs` allows mounting configurations, secrets foundin a pligos configuration to the local filesystem. This way configuration, which is typically mounted to a countainer inside Kubernetes can also be used locally. This supports local application development, by simulating the same configuration interface as inside the cluster.

jostillmanns committed:
add helmfs implementation

`pligos helmfs` allows mounting configurations, secrets foundin a
pligos configuration to the local filesystem. This way configuration,
which is typically mounted to a countainer inside Kubernetes can also be
used locally. This supports local application development, by
simulating the same configuration interface as inside the cluster.

jostillmanns committed:
apply to `single-tool` structure changes

johnholiver commented:
Got this right away
```bash
⎈  real-prod ~/Real/pligos $ pligos helmfs -m /tmp/pligos
2019/03/15 16:44:33 cannot locate OSXFUSE
```
Seems to mean that I need to install something else (a dependency) to use this command.

jostillmanns committed:
add support for dependency management

jostillmanns committed:
add support for dependencies

jostillmanns commented:
yes, you will need fuse. I have no idea how to install that on OSX.

jostillmanns committed:
refactor to use the helm in-memory representation of a chart

also: to deal with chart dependencies, this adds a new command

  pligos helm update-dependencies

that needs to be called before create to make sure all dependencies
are present.

jostillmanns committed:
implement create file handing

jostillmanns committed:
make create a sub command of helm

jostillmanns committed:
apply the changes made in the `dependency_management`

this is now
 * using the in-memory representation of a chart
 * renamed to configfs, since it's not directly related to helm, but
 only uses helm for implementation

jostillmanns committed:
fix leading `/` on config files

jostillmanns committed:
fix missing values yaml
